### PR TITLE
Refactor Recommendation code

### PR DESCRIPTION
**Pull request by jessie-ross**
**This PR refactors the `Recommendation` code in results:**
- Simplifies the code
- Allows us to use `AggregateRecommendation` in the Top Level Results

## How has this been tested?

- Automated tests that cover added/changed functionality
- Manual testing with mock data

### CODE REVIEW

**Review by withinboredom — COMMENTED** [review]
I'll wait for the other PR to pass code review, but on an initial pass: LGTM

File: src/components/experiments/single-view/results/__snapshots__/ExperimentDebug.test.tsx.snap [hunk 1]
```diff
@@ -459,8 +459,7 @@ exports[`renders the full tables with some analyses in debug mode 2`] = `
                       class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                       style="box-sizing: border-box;"
                     >
-                      Deploy 
-                      test
+                      Not analyzed yet
```
Comment (jessie-ross) [inline src/components/experiments/single-view/results/__snapshots__/ExperimentDebug.test.tsx.snap:462]: Hmm, something is wrong here.

**Review by withinboredom — APPROVED** [review]
Some minor comments and curiosities, but LGTM

File: src/components/experiments/single-view/results/FullLatestAnalyses.tsx [hunk 2]
```diff
@@ -48,8 +48,12 @@ export default function FullLatestAnalyses({
     },
     {
       title: 'Recommendation',
-      render: ({ recommendation }: Analysis) =>
-        recommendation && <RecommendationString recommendation={recommendation} experiment={experiment} />,
+      render: (analysis: Analysis) => (
+        <AggregateRecommendationDisplay
+          aggregateRecommendation={getAggregateRecommendation([analysis], analysis.analysisStrategy)}
```
[the remark this answers is not in the document]
Comment (jessie-ross) [reply src/components/experiments/single-view/results/FullLatestAnalyses.tsx:53]: I refactored the single and multiple recommendation processing code together into `getAggregateRecommendation`, I don't think we need to have two.

File: src/components/experiments/single-view/results/MetricAssignmentResults.tsx [hunk 3]
```diff
@@ -95,6 +95,10 @@ export default function MetricAssignmentResults({
     : identity
   const analyses = analysesByStrategyDateAsc[strategy]
   const latestAnalysis = _.last(analyses)
+  if (!latestAnalysis) {
+    return null
+  }
```
[the remark this answers is not in the document]
Comment (jessie-ross) [reply src/components/experiments/single-view/results/MetricAssignmentResults.tsx:100]: Yeah I thought I would try it here to keep `ActualExperimentResults` simple, not set on it though.

File: src/components/experiments/single-view/results/ActualExperimentResults.tsx [hunk 4]
```diff
@@ -253,11 +233,15 @@ export default function ActualExperimentResults({
...
         options={createStaticTableOptions(metricAssignmentSummaryData.length)}
         onRowClick={(_event, rowData, togglePanel) => {
-          const { latestDefaultAnalysis, recommendationConflict } = rowData as {
-            latestDefaultAnalysis?: Analysis
-            recommendationConflict?: boolean
+          const { aggregateRecommendation } = rowData as {
+            aggregateRecommendation: AggregateRecommendation
           }
-          if (togglePanel && latestDefaultAnalysis && !recommendationConflict) {
+          let disabled = aggregateRecommendation.decision === AggregateRecommendationDecision.ManualAnalysisRequired
```
[the remark this answers is not in the document]
Comment (jessie-ross) [reply src/components/experiments/single-view/results/ActualExperimentResults.tsx:239]: Hmm, I am not such a fan of this incarnation but in the future I might like to see some kind of permission function which could simplify this type of thing and be extensible to roles:
```
hasPermission(
  user,
  { 
    default: aggregateRecommendation.decision !== AggregateRecommendationDecision.ManualAnalysisRequired,
    experimentGuild: true,
    debugger: true,
  }
)
```

**Comment by jessie-ross** [comment]
Ugh, the base branch didn't auto-change to master 😭

**Pull request merged**